src/tools/UBGraphicsTriangle.cpp

```diff
@@ -555,9 +546,21 @@ void UBGraphicsTriangle::rotateAroundCenter(QTransform& transform, QPointF cente
 }
 
 
-QPointF    UBGraphicsTriangle::rotationCenter() const
+QPointF UBGraphicsTriangle::rotationCenter() const
 {
-    return B1;
+    QPointF center{B1};
```

Comment:
you're reverting 74570dc

Reply:
I started with the ruler: It is often used to measure some distance. Therefore I decided to use the zero of of upper scale as snap point. You can then use this to select the first point of the distance you want to measure. Then you rotate the ruler, until it goes through your second point. For this, it is essential that the zero point is the rotation center.

For the triangle I want to have the same behavior, as it also has a scale useful for measuring.

What are the arguments to use the edge of the triangle as rotation center instead?

Reply:
> Therefore I decided to use the zero of the upper scale as snap point

I'm OK with that, for the triangle too, but sill the edge of the triangle must remain the rotation center when it comes to rotating, as it was decided by the team. It's even more relevant now that the user can draw on each side of the triangle.

Reply:
> > Therefore I decided to use the zero of the upper scale as snap point
> 
> I'm OK with that, for the triangle too

Doesn't it then make sense to use the same corner used for rotation also as snap point? I still think it is not intuitive to use different points for snapping and rotating.

>  the edge of the triangle must remain the rotation center when it comes to rotating, as it was decided by the team

of course, if it is a team decision, then we have to stick to that. And for measuring distances we still have the ruler :)

Reply:
> Doesn't it then make sense to use the same corner used for rotation also as snap point? I still think it is not intuitive to use different points for snapping and rotating.

You're right. Let's use the same point for rotation as for snap points for every tool (with the edge being this point for the triangle, and the zero the one for the ruler)

> > the edge of the triangle must remain the rotation center when it comes to rotating, as it was decided by the team
> 
> of course, if it is a team decision, then we have to stick to that. And for measuring distances we still have the ruler :)

My commits, when not purely technical, are always based on what the team asked me to do :)

Reply:
I edited the commit for the triangle and reverted my changes of the rotation center. I think this was my last commit before our vacations. I will be back on 5.9.

Reply:
Thank you for all your work. Good vacations !